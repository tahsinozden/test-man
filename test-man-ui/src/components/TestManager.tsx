import React, {useEffect, useState} from "react";
import Container from "react-bootstrap/Container";
import {Col, Row} from "react-bootstrap";
import CreateTestModal from "./CreateTestModal";
import TestPreviewList from "./TestPreviewList";
import TestManagerApi from "../api/TestManagerApi";
import TestDetail from "../models/TestDetail";

const TestManager: React.FC = () => {
    const [allTests, setAllTests] = useState<TestDetail[]>([]);
    const testManagerApi = new TestManagerApi();

    const loadTestDetails = () => {
        testManagerApi.getAllTests().then(data => setAllTests(data));
    }

    useEffect(() => {
        loadTestDetails();
    }, []);

    return (
        <>
            <Container className="p-3">
                <Row>
                    <Col>
                        <CreateTestModal onNewTestCreated={loadTestDetails}/>
                    </Col>
                </Row>
            </Container>
            <Container className="p-3">
                <Row>
                    <TestPreviewList testDetails={allTests} onTestDataChange={loadTestDetails}/>
                </Row>
            </Container>
        </>
    );
}

export default TestManager;