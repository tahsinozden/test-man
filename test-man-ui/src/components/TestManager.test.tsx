import React from "react";
import {render, screen} from "@testing-library/react";
import TestManager from "./TestManager";
import renderer from "react-test-renderer";

it('renders correctly', () => {
    const tree = renderer
        .create(<TestManager/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

test("Test Manager contains correct header", () => {
    render(<TestManager/>);
    const text = screen.getByText("Test Manager");
    expect(text).toBeInTheDocument();
});

test("Test Manager contains all tests", () => {
    render(<TestManager/>);
    const text = screen.getByText("Tests");
    expect(text).toBeInTheDocument();
});