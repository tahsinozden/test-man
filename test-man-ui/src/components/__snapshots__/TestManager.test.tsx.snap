// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <ul
    className="ant-menu-overflow ant-menu ant-menu-root ant-menu-horizontal ant-menu-dark"
    data-menu-list={true}
    onKeyDown={[Function]}
    role="menu"
    tabIndex={0}
  >
    <li
      className="ant-menu-overflow-item ant-menu-item ant-menu-item-only-child"
      data-menu-id="rc-menu-uuid-test-mainPage"
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="menuitem"
      style={
        Object {
          "height": undefined,
          "opacity": 1,
          "order": 0,
          "overflowY": undefined,
          "pointerEvents": undefined,
          "position": undefined,
        }
      }
      tabIndex={-1}
    >
      <span
        className="ant-menu-title-content"
      >
        Test Manager
      </span>
    </li>
    <li
      aria-hidden={true}
      className="ant-menu-overflow-item ant-menu-overflow-item-rest ant-menu-submenu ant-menu-submenu-horizontal ant-menu-submenu-disabled"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="none"
      style={
        Object {
          "height": 0,
          "opacity": 0,
          "order": 9007199254740991,
          "overflowY": "hidden",
          "pointerEvents": "none",
          "position": "absolute",
        }
      }
    >
      <div
        aria-controls="rc-menu-uuid-test-rc-menu-more-popup"
        aria-disabled={true}
        aria-expanded={false}
        aria-haspopup={true}
        className="ant-menu-submenu-title"
        data-menu-id="rc-menu-uuid-test-rc-menu-more"
        onClick={[Function]}
        onFocus={[Function]}
        role="menuitem"
        style={null}
        tabIndex={null}
        title={null}
      >
        <span
          aria-label="ellipsis"
          className="anticon anticon-ellipsis"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="ellipsis"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
            />
          </svg>
        </span>
        <i
          className="ant-menu-submenu-arrow"
        />
      </div>
    </li>
  </ul>,
  <div
    aria-hidden={true}
    style={
      Object {
        "display": "none",
      }
    }
  />,
  <section
    className="ant-layout"
  >
    <main
      className="ant-layout-content"
    >
      <div
        className="ant-divider ant-divider-horizontal"
        role="separator"
      />
      <div
        className="ant-row"
        style={Object {}}
      >
        <div
          className="ant-col ant-col-4"
          style={Object {}}
        />
        <div
          className="ant-col"
          style={Object {}}
        >
          <button
            className="ant-btn ant-btn-primary"
            onClick={[Function]}
            type="button"
          >
            <span>
              Create a new test
            </span>
          </button>
        </div>
      </div>
      <div
        className="ant-divider ant-divider-horizontal"
        role="separator"
      />
      <div
        className="ant-row"
        style={Object {}}
      >
        <div
          className="ant-col ant-col-4"
          style={Object {}}
        />
        <div
          className="ant-col"
          style={Object {}}
        >
          <h3>
            Tests
          </h3>
        </div>
      </div>
      <div
        className="ant-row"
        style={Object {}}
      >
        <div
          className="ant-col ant-col-4"
          style={Object {}}
        />
        <div
          className="ant-col"
          style={Object {}}
        >
          <section
            className="ant-layout"
          />
        </div>
      </div>
      <div
        className="ant-divider ant-divider-horizontal"
        role="separator"
      />
    </main>
  </section>,
]
`;
